### PR TITLE
Fix nullable references

### PR DESCRIPTION
**Pull request by mikeebowen**
This pull request fixes nullable reference errors in the samples and makes them executable with command line arguments.

### CODE REVIEW

**Comment by mikeebowen** [comment]
@twsouthwick, please review and make sure everything  looks OK.

File: samples/presentation/create_by_providing_a_file_name/cs/Program.cs [hunk 1]
```diff
@@ -1,294 +1,284 @@
...
+    PresentationDocument presentationDoc = PresentationDocument.Create(filepath, PresentationDocumentType.Presentation);
+    PresentationPart presentationPart = presentationDoc.AddPresentationPart();
+    presentationPart.Presentation = new Presentation();
 
-            //Dispose the presentation handle
-            presentationDoc.Dispose();
-        }
+    CreatePresentationParts(presentationPart);
 
-        private static void CreatePresentationParts(PresentationPart presentationPart)
```
Comment (twsouthwick) [inline samples/presentation/create_by_providing_a_file_name/cs/Program.cs:32]: why is private removed?

File: samples/presentation/retrieve_the_number_of_slides/cs/Program.cs [hunk 2]
```diff
@@ -1,37 +1,49 @@
...
 using System.Linq;
 
+if (args.Length >= 2)
+{
+    RetrieveNumberOfSlides(args[0], args[1]);
+}
+else
+{
+    RetrieveNumberOfSlides(args[0]);
+}
```
Comment (twsouthwick) [inline samples/presentation/retrieve_the_number_of_slides/cs/Program.cs:12]:
```suggestion
if (args is [ {} fileName, {} includeHidden])
{
    RetrieveNumberOfSlides(fileName, includeHidden);
}
else if (args is [ {} fileName ])
{
    RetrieveNumberOfSlides(fileName);
}
```

not needed in this PR, but using pattern matching will clean the samples up a bit and do better error checking

Comment (twsouthwick) [reply samples/presentation/retrieve_the_number_of_slides/cs/Program.cs:12]: also, in a later pr, strings should be pre-parsed for the methods so the methods (such as RetrieveNumberOfSlides) takes the types it actually needs

**Review by twsouthwick — APPROVED** [review]
LGTM - a few thoughts for patterns me want to do for later

File: samples/spreadsheet/insert_a_new_worksheet/cs/Program.cs [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
     {
+        WorkbookPart workbookPart = spreadSheet.WorkbookPart ?? spreadSheet.AddWorkbookPart();
         // Add a blank WorksheetPart.
-        WorksheetPart newWorksheetPart = spreadSheet.WorkbookPart.AddNewPart<WorksheetPart>();
+        WorksheetPart newWorksheetPart = workbookPart.AddNewPart<WorksheetPart>();
         newWorksheetPart.Worksheet = new Worksheet(new SheetData());
 
-        Sheets sheets = spreadSheet.WorkbookPart.Workbook.GetFirstChild<Sheets>();
-        string relationshipId = spreadSheet.WorkbookPart.GetIdOfPart(newWorksheetPart);
+        Sheets sheets = workbookPart.Workbook.GetFirstChild<Sheets>() ?? workbookPart.Workbook.AppendChild(new Sheets());
```
Comment (twsouthwick) [inline samples/spreadsheet/insert_a_new_worksheet/cs/Program.cs:18]: should we create a new API here for `GetFirstChildOrCreate<TPart>()`?

File: samples/word/create_and_add_a_paragraph_style/cs/Program.cs [hunk 4]
```diff
@@ -1,81 +1,89 @@
...
+
+            // Add the style to the styles part.
+            styles.Append(style);
+        }
+    }
 }
 
 // Add a StylesDefinitionsPart to the document.  Returns a reference to it.
-static StyleDefinitionsPart AddStylesPartToPackage(WordprocessingDocument doc)
+static StyleDefinitionsPart AddStylesPartToPackage(WordprocessingDocument? doc)
```
Comment (twsouthwick) [inline samples/word/create_and_add_a_paragraph_style/cs/Program.cs:78]:
```suggestion
static StyleDefinitionsPart AddStylesPartToPackage(WordprocessingDocument doc)
```

why not change it to require non-nullable?

File: samples/spreadsheet/insert_a_new_worksheet/cs/Program.cs (hunk 3 above)
Comment (mikeebowen) [reply samples/spreadsheet/insert_a_new_worksheet/cs/Program.cs:18]: That would be a more clear than `Sheets sheets = workbookPart.Workbook.GetFirstChild<Sheets>() ?? workbookPart.Workbook.AppendChild(new Sheets());`

File: samples/word/retrieve_comments/cs/Program.cs [hunk 5]
```diff
@@ -1,18 +1,21 @@
...
 {
-    using (WordprocessingDocument wordDoc =
-        WordprocessingDocument.Open(fileName, false))
+    using (WordprocessingDocument wordDoc = WordprocessingDocument.Open(fileName, false))
     {
-        WordprocessingCommentsPart commentsPart =
-            wordDoc.MainDocumentPart.WordprocessingCommentsPart;
+        if (wordDoc.MainDocumentPart is null || wordDoc.MainDocumentPart.WordprocessingCommentsPart is null)
+        {
+            throw new System.NullReferenceException("MainDocumentPart and/or WordprocessingCommentsPart is null.");
```
Comment (twsouthwick) [inline samples/word/retrieve_comments/cs/Program.cs:13]: instead of nullreferenceexception, should throw InvalidOperation

**Review by twsouthwick — CHANGES_REQUESTED** [review]
throwing nullreferenceexception is probably not the right exception to throw when checking - maybe invalidoperationexception?

File: samples/word/set_a_custom_property/cs/Program.cs [hunk 6]
```diff
@@ -44,7 +45,7 @@
         case PropertyTypes.NumberInteger:
             if ((propertyValue) is int)
             {
-                newProp.VTInt32 = new VTInt32(propertyValue.ToString());
+                newProp.VTInt32 = new VTInt32(propertyValueString);
```
Comment (twsouthwick) [inline samples/word/set_a_custom_property/cs/Program.cs:48]: why this change?

File: samples/word/create_and_add_a_paragraph_style/cs/Program.cs (hunk 4 above)
Comment (mikeebowen) [reply samples/word/create_and_add_a_paragraph_style/cs/Program.cs:78]: Because that way I can write:
```csharp
Styles? styles = wordprocessingDocument?.MainDocumentPart?.StyleDefinitionsPart?.Styles ?? AddStylesPartToPackage(wordprocessingDocument).Styles;
```
insted of 
```csharp
Styles? styles = wordprocessingDocument?.MainDocumentPart?.StyleDefinitionsPart?.Styles;

if (wordprocessingDocument is not null) {
    Styles styles = AddStylesPartToPackage(wordprocessingDocument).Styles;
    // do stuff...
}
```

Comment (twsouthwick) [reply samples/word/create_and_add_a_paragraph_style/cs/Program.cs:78]: Looks like we may want to look at some new APIs to enable something like:

```csharp
Styles styles = wordprocessingDocument.GetOrCreate<MainDocumentPart>().GetOrCreate<StyleDefinitionsPart>().Styles
```

Comment (twsouthwick) [reply samples/word/create_and_add_a_paragraph_style/cs/Program.cs:78]: I'd suggest tracking any of these "convoluted" rewritings you have to do so we can identify some new APIs to help

File: samples/word/set_a_custom_property/cs/Program.cs (hunk 6 above)
Comment (mikeebowen) [reply samples/word/set_a_custom_property/cs/Program.cs:48]: I don't know @tomjebo made that change

File: samples/presentation/create_by_providing_a_file_name/cs/Program.cs (hunk 1 above)
Comment (mikeebowen) [reply samples/presentation/create_by_providing_a_file_name/cs/Program.cs:32]: Because with private set I get [CS0106](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0106?f1url=%3FappId%3Droslyn%26k%3Dk(CS0106)): The modifier 'private' is not valid for this item" error from Visual Studio if is set to private.

File: samples/presentation/retrieve_the_number_of_slides/cs/Program.cs (hunk 2 above)
Comment (mikeebowen) [reply samples/presentation/retrieve_the_number_of_slides/cs/Program.cs]: Made this change since I was making other changes.

File: samples/word/retrieve_comments/cs/Program.cs (hunk 5 above)
Comment (mikeebowen) [reply samples/word/retrieve_comments/cs/Program.cs]: Done

File: samples/spreadsheet/insert_a_new_worksheet/cs/Program.cs (hunk 3 above)
Comment (twsouthwick) [reply samples/spreadsheet/insert_a_new_worksheet/cs/Program.cs:18]: https://github.com/dotnet/Open-XML-SDK/issues/1548

File: samples/word/set_a_custom_property/cs/Program.cs [hunk 7]
```diff
@@ -106,10 +107,10 @@
             // This will trigger an exception if the property's Name 
             // property is null, but if that happens, the property is damaged, 
             // and probably should raise an exception.
+#pragma warning disable CS8602 // Dereference of a possibly null reference.
```
Comment (twsouthwick) [inline samples/word/set_a_custom_property/cs/Program.cs:110]: ? do we still need this? Maybe just use the `!` operator?

File: samples/word/set_the_font_for_a_text_run/cs/Program.cs [hunk 8]
```diff
@@ -18,6 +18,11 @@ static void SetRunFont(string fileName)
                 Ascii = "Arial"
             });
 
+        if (package.MainDocumentPart is null)
+        {
+            throw new System.NullReferenceException("MainDocumentPart is null.");
```
Comment (twsouthwick) [inline samples/word/set_the_font_for_a_text_run/cs/Program.cs:23]: These should be removed and replaced with ArgumentNullException or InvalidOperationException throughout these changes

**Review by twsouthwick — CHANGES_REQUESTED** [review]

File: samples/word/create_and_add_a_paragraph_style/cs/Program.cs (hunk 4 above)
Comment (twsouthwick) [reply samples/word/create_and_add_a_paragraph_style/cs/Program.cs:78]: See https://github.com/dotnet/Open-XML-SDK/issues/1548 for an issue with a few examples we can continue the conversation on

**Comment by twsouthwick** [comment]
@mikeebowen I think the only *blocking* thing here is to remove the manual throwing of the NullReferenceExceptions. It would be nice to add `using System.` so we don't end up with `System.ArgumentNullException` as well

**Comment by mikeebowen** [comment]
Do you want to change all the NullReferenceExceptions to ArgumentNullException? Many of them are not null arguments, but possibly null parts and elements of the document.

**Review by lindalu-MSFT — APPROVED** [review]

File: samples/word/set_a_custom_property/cs/Program.cs (hunk 6 above)
Comment (tomjebo) [reply samples/word/set_a_custom_property/cs/Program.cs:48]: the argument was used several times in series so I thought it better to test it once and throw if null.

**Comment by twsouthwick** [comment]
> Do you want to change all the NullReferenceExceptions to ArgumentNullException? Many of them are not null arguments, but possibly null parts and elements of the document.

It is a big code smell to throw NullReferenceExceptions - you are supposed to throw ArgumentNullExceptions if you have something that should not be null but is. If you don't think that's appropriate, then you'll want to refactor things so that you don't need to throw either.

**Comment by lindalu-MSFT** [comment]
@twsouthwick Please ping me when all comments are resolved and you're ready to merge this PR! Thanks, Linda

File: samples/word/accept_all_revisions/cs/Program.cs [hunk 9]
```diff
@@ -66,7 +67,7 @@ static void AcceptAllRevisions(string fileName, string authorName)
                 }
                 else
                 {
-                    OpenXmlElement? nextSibling = insertion.NextSibling() ?? throw new System.NullReferenceException("NextSibling is null.");
+                    OpenXmlElement? nextSibling = insertion.NextSibling() ?? throw new ArgumentNullException("NextSibling is null.");
```
Comment (twsouthwick) [inline samples/word/accept_all_revisions/cs/Program.cs:70]: nextSibling will be non nullable here

File: samples/word/change_the_print_orientation/cs/Program.cs [hunk 10]
```diff
@@ -37,7 +38,7 @@ static void SetPrintOrientation(string fileName, string no)
         {
             bool pageOrientationChanged = false;
 
-            PageSize pgSz = (sectPr.Descendants<PageSize>().FirstOrDefault()) ?? throw new System.NullReferenceException("There are no PageSize elements in the section.");
+            PageSize pgSz = (sectPr.Descendants<PageSize>().FirstOrDefault()) ?? throw new ArgumentNullException("There are no PageSize elements in the section.");
```
Comment (twsouthwick) [inline samples/word/change_the_print_orientation/cs/Program.cs:41]: Change this to `.First()` instead of throwing exceptions - it'll ensure there is at least one

File: samples/word/accept_all_revisions/cs/Program.cs (hunk 9 above)
Comment (mikeebowen) [reply samples/word/accept_all_revisions/cs/Program.cs:70]: `OpenXmlElement.NextSibling()` returns `OpenXmlElement?`

**Review by twsouthwick — APPROVED** [review]
LGTM thanks for responding to the feedback!

@lindalu-MSFT this can be merged now

**Comment by twsouthwick** [comment]
@lindalu-MSFT is there a staging build we can use to see the final output as we work through the items in #159 ?

**Comment by lindalu-MSFT** [comment]
> @lindalu-MSFT is there a staging build we can use to see the final output as we work through the items in #159 ?

I don't know about .cs code, but for docs, you go to the live location you want to review the changes and insert "review." at the front of the URL. For example, https://learn.microsoft.com/en-us/office/open-xml/open-xml-sdk becomes https://review.learn.microsoft.com/en-us/office/open-xml/open-xml-sdk.

**Comment by twsouthwick** [comment]
> > @twsouthwick  is there a staging build we can use to see the final output as we work through the items in #159 ?
> 
> I don't know about .cs code, but for docs, you go to the live location you want to review the changes and insert "review." at the front of the URL. For example, https://learn.microsoft.com/en-us/office/open-xml/open-xml-sdk becomes https://review.learn.microsoft.com/en-us/office/open-xml/open-xml-sdk.

Perfect! The code gets pulled in via includes, so they'll show up there and we can validate things actually look right (we can only get so far with the docfx cli)

**Comment by lindalu-MSFT** [comment]
@twsouthwick Please resolve these merge conflicts then I will merge to live.
https://github.com/OfficeDev/open-xml-docs/pull/174
